File: leetcode/trie/LongestCommonPrefix.java
/*
14. Longest Common Prefix
Solved
Easy
Topics
Companies
Write a function to find the longest common prefix string amongst an array of strings.

If there is no common prefix, return an empty string "".
*/

class LongestCommonPrefix {
    // 0ms Beats 100%
    public String longestCommonPrefix(String[] strs) {
        int numStrings = strs.length;

        String prefix = strs[0];

        int i = 1;
        while(i < strs.length){
            while(!strs[i].startsWith(prefix))
                prefix = prefix.substring(0, prefix.length() -1);
            i++;
        }
        return prefix;
    }
}

File: leetcode/strings/two_pointers/longest_common_prefix.py
"""
14. Longest Common Prefix
Solved
Easy
Topics
Companies
Write a function to find the longest common prefix string amongst an array of strings.

If there is no common prefix, return an empty string "".
"""


class LongestCommonPrefix:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        # Iterate through prefixes from first string
        for i in range(len(strs[0])):
            # Compare it with all the strings, if mismatch return
            for s in strs:
                if i == len(s) or s[i] != strs[0][i]:
                    return strs[0][:i]

        return strs[0]
